src/NUnitEngine/nunit.engine/Services/RuntimeFrameworkService.cs

```diff
@@ -109,21 +129,45 @@ private static RuntimeFramework SelectRuntimeFrameworkInner(TestPackage package)
                     throw new NUnitEngineException("Invalid or unknown framework requested: " + frameworkSetting);
 
                 log.Debug($"Requested framework for {package.Name} is {requestedFramework}");
+
+                if (!IsAvailable(frameworkSetting))
+                    throw new NUnitEngineException("Requested framework is not available: " + frameworkSetting);
+
+                package.Settings[EnginePackageSettings.TargetRuntimeFramework] = frameworkSetting;
+
+                return requestedFramework;
             }
-            else
+
+            log.Debug($"No specific framework requested for {package.Name}");
+
+            string imageTargetFrameworkNameSetting = package.GetSetting(InternalEnginePackageSettings.ImageTargetFrameworkName, "");
+            RuntimeType targetRuntime;
+            Version targetVersion;
+
+            if (string.IsNullOrEmpty(imageTargetFrameworkNameSetting))
             {
-                requestedFramework = new RuntimeFramework(RuntimeType.Any, RuntimeFramework.DefaultVersion);
-                log.Debug($"No specific framework requested for {package.Name}");
+                // Assume .NET Framework
+                targetRuntime = RuntimeType.Net;
+                targetVersion = package.GetSetting(InternalEnginePackageSettings.ImageRuntimeVersion, new Version(2, 0));
```

Comment:
Niggle - this looks to be a logic change when ImageTargetFrameworkName is unset - previously we defaulted to the current framework, now we default to net20.

Not totally averse to the change if there's a particular reason for it, but I feel like the old behaviour was preferable. We know the current framework exists, and if the console has launched under it, then it's likely to be the "default" framework on this machine for other .NET executables. It seems to make sense to run the tests on the default framework, if we have no better information to go off.

Afraid I don't know how often ImageTargetFrameworkName would be unset, to judge how much of an edge-case this is that we're talking about here!

Reply:
I made the change because I've only seen it unset under pre-4.0 frameworks, using old versions of VS. CurrentFramework is safer, especially if we eventually add this logic to a .NET Core runner. I'll try it and we can see what it does to the tests.

Reply:
> especially if we eventually add this logic to a .NET Core runner

That hadn't crossed my mind, and is a really good point! Thanks. 🙂

Reply:
I think this change is still outstanding too, Charlie.

Reply:
Yes, I guess it still makes sense to use Current framework in 3.x, since `--inprocess` is still supported.